[Wallet] Upgrade sentry/react-native to 2.1.0

etuleu opened the pull request:
### Description

Upgrade sentry/react-native from 1.7.1 to 2.1.0.

This is part of our effort to improve debugging capabilities. With this update we hope to see better stack traces in some native crashes where the entire trace is missing.

I went through the release notes for each of the releases since 1.7.1 to make sure we capture any breaking changes and to see what has been fixed and what to watch out for.

One thing that has been added is release health which we should discuss whether we want on or off: https://blog.sentry.io/2020/05/19/crash-free-sessions-carefree-users-with-release-health

What's new in sentry:

- Bump sentry-android to 3.2.0 https://github.com/getsentry/sentry-react-native/pull/1208
- Fix: set userId for hard-crashes if no user is set [#1049](https://github.com/getsentry/sentry-java/pull/1049) 
- Bump @sentry/javascript dependencies to [5.28.0](https://github.com/getsentry/sentry-react-native/pull/1228)
- Bump sentry-cocoa to [6.0.9](https://github.com/getsentry/sentry-react-native/issues/1200)


### Other changes
Set sentry automatic session tracking explicitly off because the default has changed in 2.0. Not sure yet if we want that on so I set it off to be explicit.


### Tested

Manually causing crashes in JS:
`throw new Error("My first Sentry error!");`
and in native:
`Sentry.nativeCrash();`

### Related issues

- Fixes #6203

### Backwards compatibility

Yes.

etuleu committed:
upgrade sentry/react-native to 2.1.0

etuleu committed:
Merge branch 'master' into etuleu/upgrade-sentry-2.1.0

etuleu committed:
updated podfile.lock

etuleu committed:
update sentry package name for android

etuleu committed:
disable enableAutoSessionTracking

jeanregisser commented on packages/mobile/src/sentry/Sentry.ts:
```diff
@@ -13,7 +13,11 @@ export async function installSentry() {
+  Sentry.init({
+    dsn: SENTRY_URL,
+    environment: DeviceInfo.getBundleId(),
+    enableAutoSessionTracking: false,
```
Reading up on this, I think we should enable it. https://docs.sentry.io/product/releases/health/
```suggestion
    enableAutoSessionTracking: true,
```

jeanregisser reviewed (approved):
Looking great! 👍

jeanregisser committed:
Fix test failure caused by using different packages due to yarn hoisting

@apollo/react-testing ended up being hoisted in the root node_modules
but the other @apollo libs used by react-apollo were still in packages/mobile/node_modules
so the created context didn't match and failed in `TransactionsList.test.tsx`
with the following error:
```
Invariant Violation: Could not find "client" in the context or passed in as an option. Wrap the root component in an <ApolloProvider>, or pass an ApolloClient instance in via options.
```

etuleu committed:
Update packages/mobile/src/sentry/Sentry.ts

Co-authored-by: Jean Regisser <jean.regisser@gmail.com>

etuleu committed:
Merge branch 'master' into etuleu/upgrade-sentry-2.1.0

mergify[bot] committed:
Merge branch 'master' into etuleu/upgrade-sentry-2.1.0

mergify[bot] committed:
Merge branch 'master' into etuleu/upgrade-sentry-2.1.0

mergify[bot] committed:
Merge branch 'master' into etuleu/upgrade-sentry-2.1.0

mergify[bot] committed:
Merge branch 'master' into etuleu/upgrade-sentry-2.1.0

mergify[bot] committed:
Merge branch 'master' into etuleu/upgrade-sentry-2.1.0

mergify[bot] committed:
Merge branch 'master' into etuleu/upgrade-sentry-2.1.0

Lss-Ankit commented:
@etuleu  can you please provide more details if we need to verify this task from our side.
